cmd/cloudtrail/write-events.go

```diff
@@ -0,0 +1,392 @@
+
+	return &LookupEventsoptions{}
+
+}
+
+// complete fills in necessary data for the options struct
+func (o *LookupEventsoptions) complete(cmd *cobra.Command, _ []string) error {
+	err := utils.IsValidClusterKey(o.clusterID)
+	if err != nil {
+		return err
+	}
+
+	connection, err := utils.CreateConnection()
+	if err != nil {
+		return err
+	}
+
+	defer connection.Close()
+
+	cluster, err := utils.GetCluster(connection, o.clusterID)
+	if err != nil {
+		return err
+	}
+
+	o.cluster = cluster
+	o.clusterID = cluster.ID()
+
+	if strings.ToUpper(cluster.CloudProvider().ID()) != "AWS" {
+		return errors.New("this command is only available for AWS clusters")
+	}
+
+	return nil
+}
+
+// ParseDurationToUTC parses a duration string and returns a UTC time
+func ParseDurationToUTC(input string) (time.Time, error) {
+
+	duration, err := time.ParseDuration(input)
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	return time.Now().UTC().Add(-duration), nil
+}
+
+// Whoami retrieves caller identity information
+func Whoami(stsClient sts.Client) (Arn string, AccountId string, err error) {
+
+	ctx := context.TODO()
+	callerIdentityOutput, err := stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
+	if err != nil {
+		return "", "", err
+	}
+	userArn, err := arn.Parse(*callerIdentityOutput.Arn)
+	if err != nil {
+		return "", "", err
+	}
+
+	return userArn.String(), userArn.AccountID, nil
+}
+
+// GetEvents retrieves cloudtrail events since the specified time
+// using the provided cloudtrail client and starttime from since flag.
+func GetEvents(since time.Time, client *cloudtrail.Client, pages int) ([]*cloudtrail.LookupEventsOutput, error) {
+
+	ctx := context.TODO()
+	starttime := since
+	cloudtrailClient := client
+
+	lookupEventlookupOutputs := []*cloudtrail.LookupEventsOutput{}
+	var input = cloudtrail.LookupEventsInput{
+		StartTime: &starttime,
+		EndTime:   aws.Time(time.Now()),
+		LookupAttributes: []types.LookupAttribute{
+			{AttributeKey: "ReadOnly",
+				AttributeValue: aws.String("false")},
+		},
+	}
+
+	maxPages := pages
+
+	for counter := 0; counter <= maxPages; counter++ {
+
+		lookupOutput, err := cloudtrailClient.LookupEvents(ctx, &input)
+		if err != nil {
+			return nil, err
+		}
+
+		lookupEventlookupOutputs = append(lookupEventlookupOutputs, lookupOutput)
+
+		input.NextToken = lookupOutput.NextToken
+		if lookupOutput.NextToken == nil {
+			break
+		}
+
+	}
+
+	return lookupEventlookupOutputs, nil
+}
+
+// Extracts Raw cloudtrailEvent Details
+func ExtractUserDetails(cloudTrailEvent *string) (*RawEventDetails, error) {
+	if cloudTrailEvent == nil || *cloudTrailEvent == "" {
+		return &RawEventDetails{}, fmt.Errorf("cannot parse a nil input")
+	}
+	var res RawEventDetails
+	err := json.Unmarshal([]byte(*cloudTrailEvent), &res)
+	if err != nil {
+		return &RawEventDetails{}, fmt.Errorf("could not marshal event.CloudTrailEvent: %w", err)
+	}
+	supportedEventVersions := []string{"1.08", "1.09", "1.10"}
```

Comment:
**Suggestion:** We can decide whether we support the version in a much better way. Take a look here: https://github.com/openshift/configuration-anomaly-detection/pull/240/files